### ISSUE TITLE
Update client config `tracked_shards`

### ISSUE THREAD

**Issue by mzhangmzz** [near/nearcore#4930]
With the simple nightshade sharding upgrade, the `tracked_shards` field in client config no longer makes sense.

First a client should not care about a particular shard at all. This also makes the interface more robust to sharding changes.

Second, the meaning of shard_id changes when resharding happens. For example, if a client starts with tracking shard 0 when there is one shard, and the shard is split to eight shards, should this client keep track shard 0 or track all shards? We would think the latter. But the client config will not change. If the client restarts with the same config, it will track shard 0 instead of all shards. The client needs to update their config every time when resharding happens. This is very complicated and unnecessary.

[https://github.com/near/nearcore/pull/4668](https://github.com/near/nearcore/pull/4668) changes ShardTracker and NightshadeRuntime to remove the option of `tracked_shards` and added `track_all_shards` instead. However, it doesn't update client config format because of backwards compatibility. This issue tracks that since we should update client config format as well.

**Comment by khorolets** [comment]
We had another conversation with @mzhangmzz and accidentally found out that the Node Interfaced team has a bit different plans (not strict though). We were considering running separate instances of NEAR Indexer for Explorer for each shard to manage the data flow. 

But the approach described in this issue expects the node to track an account (without even knowing on which shard the account is living). 

/cc @frol

**Comment by bowenwang1996** [comment]
@khorolets ah that is very interesting! I think there may be a use case for tracking a specific shard by id after all :)

**Comment by frol** [comment]
Yeah, we need to design the solution for the use-cases when you need to actually track the whole network, but cannot do it on a single physical machine anymore. Indexer is just one of the examples, but RPC services could also need it.
